src/net_processing.cpp

```diff
@@ -2593,6 +2593,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         if (pfrom.m_tx_relay != nullptr) {
             LOCK(pfrom.m_tx_relay->cs_filter);
             pfrom.m_tx_relay->fRelayTxes = fRelay; // set to true after we get the first filter* message
+            if (fRelay) pfrom.m_relays_txs = true;
```

Comment:
Because of the comment for m_relay_txs in net.h:

    /** ... This only changes from false to true. It will never change
     *  back to false. Used only in inbound eviction logic. */

This doesn't work:

```suggestion
            pfrom.m_relays_txs &= fRelay;
```

since the `&=` operator isn't defined for `std::atomic<bool>`. I could use:

```suggestion
            pfrom.m_relays_txs = pfrom.m_relay_txs && fRelay;
```

but I don't think that's an improvement.

Reply:
`pfrom.m_relays_txs = pfrom.m_relay_txs && fRelay;` could change it from `true` to `false`.

Reply:
I think the comment is a good justification to keeping the code this way.
The only problem I can see is the temptation of future devs to change this.

Reply:
> pfrom.m_relays_txs = pfrom.m_relay_txs && fRelay; could change it from true to false.

:man_facepalming: You're right of course. I meant `pfrom.m_relay_txs || fRelay;`

Reply:
> Can you move this out of the lock scope to make that clear?

Done in #24692